src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs

```diff
@@ -31,19 +32,21 @@ internal BufferBasedStatsDPublisher(StatsDConfiguration configuration, IStatsDTr
             _formatter = new StatsDUtf8Formatter(configuration.Prefix);
         }
 
-        public void Increment(long value, double sampleRate, string bucket)
+        public void Increment(long value, double sampleRate, string bucket, IDictionary<string, string>? tags = null)
         {
-            SendMessage(sampleRate, StatsDMessage.Counter(value, bucket));
+            SendMessage(sampleRate, StatsDMessage.Counter(value, bucket, tags));
         }
 
-        public void Gauge(double value, string bucket)
+        public void Gauge(double value, string bucket,
+            Operation operation = Operation.Set,
+            IDictionary<string, string>? tags = null)
```

Comment:
Nit: one parameter per line if not on the same line.

```suggestion
        public void Gauge(
            double value,
            string bucket,
            Operation operation = Operation.Set,
            IDictionary<string, string>? tags = null)
```